### PR TITLE
[DOCS] Update scripted upsert example

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the scripted upsert example to use a script `source` rather than
a stored script `id`. This makes the example more copy/paste-able.

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/docs/update.asciidoc [hunk 1]
```diff
@@ -331,23 +331,24 @@ To run the script whether or not the document exists, set `scripted_upsert` to
...
-        "time": "2014-01-01 12:32"
+    "source": """
+      if ( ctx.op == 'create' ) {
+        ctx._source.counter = params.count
+      } else {
+        ctx._source.counter += params.count
       }
+    """,
+    "params": {
+      "count": 4
```
Comment (jrodewig) [inline docs/reference/docs/update.asciidoc:346]: @jdconrad I tested this locally, but can you give me a quick check for style/syntax?

**Review by jdconrad — APPROVED** [review]
LGTM! Apologies for the delay in review.

**Comment by droberts195** [comment]
@jrodewig this PR was backported at an unfortunate time with respect to the 7.17 branching.  It's ended up in 7.15 and 7.16 but not 7.17.  Please could you merge it to 7.17 too?

**Comment by jrodewig** [comment]
Thanks for noticing this @droberts195. I've opened a backport with https://github.com/elastic/elasticsearch/pull/81857.

**Pull request merged**